grace/server.go

```diff
@@ -240,6 +248,7 @@ func (srv *Server) serverTimeout(d time.Duration) {
 			break
 		}
 		srv.wg.Done()
+		runtime.Gosched()
```

Comment:
It is a corner case, if we don't goshced here, then we have a change to loop again without srv.state been changed to StateTerminate, this race condition will make the waitGroup step == -1, that's a panic.